test/lib/find-scaffolds.js

```diff
@@ -0,0 +1,134 @@
+import test from 'blue-tape';
+import npm from 'npm';
+import {stub, spy} from 'sinon';
+
+
+import findScaffolds from '../../lib/find-scaffolds';
+
+
+const dependencies = {
+  test: {
+    // should skip this one
+    name: 'whatever-random-name',
+    realPath: 'test realpath'
+    // no keywords, should not fail
+  },
+  prefixedName: {
+    // should return, because it matches name = 'cf-*'
+    name: 'cf-test',
+    realPath: 'prefixed realpath',
+    keywords: ['random', 'keywords']
+  },
+  taggedWithSingleTag: {
+    // should return, because it matches tag 'cloverfield-scaffold'
+    name: 'tagged1',
+    realPath: 'tagged1 realpath',
+    keywords: ['cloverfield-scaffold']
+  },
+  taggedWithTwoTags: {
+    // should return, because it matches two tags 'cloverfield' && 'scaffold'
+    name: 'tagged2',
+    realPath: 'tagged2 realpath',
+    keywords: ['cloverfield', 'scaffold']
+  },
+  incorrectlyTagged: {
+    // should skip this because it matches only one tag 'cloverfield', but not 'scaffold'
+    name: 'incorrectly-tagged',
+    realPath: 'incorrectly-tagged realpath',
+    keywords: ['cloverfield', 'something else']
+  }
+};
+
+
+const before = () => {
+  // Stubs
+  const load = spy();
+
+  stub(npm, 'load', load);
+  npm.commands = {
+    ls: spy()
+  };
+
+  return {load};
+};
+
+
+const after = () => {
+  npm.load.restore();
+};
+
+
+test('Find Scaffolds success', assert => {
+
+  const {load} = before();
+
+  assert.ok(findScaffolds instanceof Function, 'should be function');
+
+
+  const promise = findScaffolds();
+
+  assert.ok(promise instanceof Promise, 'should return Promise');
+
+
+  assert.ok(load.calledOnce, 'npm.init should be called');
+
+
+  // Emulate successfull load response
+  const loadSuccess = load.getCall(0).args[1];
+
+  loadSuccess(null);
+
+  assert.ok(npm.commands.ls.calledOnce, 'npm.commands.ls should be called');
+
+
+  const lsSuccess = npm.commands.ls.getCall(0).args[2];
+
+  lsSuccess(null, {dependencies});
+
+  return promise
+    .then(scaffolds => {
+      assert.ok(scaffolds, 'find-scaffold should resolve with found scaffolds');
+      assert.deepEqual(Object.keys(scaffolds), ['test', 'tagged1', 'tagged2'],
+        'should return correctly matched scaffolds');
+    })
+    .then(after);
+});
+
+
+test('Find Scaffolds fail on npm.init', assert => {
+  const {load} = before();
+  const promise = findScaffolds();
+  const loadError = load.getCall(0).args[1];
```

Comment:
This variables name is confusing to me. To me, it makes more sense to consistently name this variable `loaded` since that is what `arg[1]` is. It only becomes an error after passing an error to the function.

loadError -> loaded
loadSuccess -> loaded